cmd/process-agent/command/command.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package command implements the top-level `process-agent` binary, including its subcommands.
+package command
+
+import (
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+
+	"github.com/DataDog/datadog-agent/cmd/process-agent/flags"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	pconfig "github.com/DataDog/datadog-agent/pkg/process/config"
+)
+
+const LoggerName config.LoggerName = "PROCESS"
+
+// GlobalParams contains the values of agent-global Cobra flags.
+//
+// A pointer to this type is passed to SubcommandFactory's, but its contents
+// are not valid until Cobra calls the subcommand's Run or RunE function.
+type GlobalParams struct {
+	// ConfFilePath holds the path to the folder containing the configuration
+	// file, to allow overrides from the command line
+	ConfFilePath string
+
+	// SysProbeConfFilePath holds the path to the folder containing the system-probe
+	// configuration file, to allow overrides from the command line
+	SysProbeConfFilePath string
+
+	// PidFilePath specifies the path to the pid file
+	PidFilePath string
+
+	// Info
+	Info bool
+
+	// WinParams provides windows specific options
+	WinParams WinParams
+}
+
+// WinParams specifies Windows-specific CLI params
+type WinParams struct {
+	// StartService handles starting the service
+	StartService bool
+
+	// StopService handles stopping the service
+	StopService bool
+
+	// Foreground handles running the service in the foreground
+	Foreground bool
+}
+
+// SubcommandFactory is a callable that will return a slice of subcommands.
+type SubcommandFactory func(globalParams *GlobalParams) []*cobra.Command
+
+// MakeCommand makes the top-level Cobra command for this app.
+func MakeCommand(subcommandFactories []SubcommandFactory, winParams bool, rootCmdRun func(globalParams *GlobalParams)) *cobra.Command {
+	globalParams := GlobalParams{}
+
+	rootCmd := &cobra.Command{
+		Run: func(cmd *cobra.Command, args []string) {
+			rootCmdRun(&globalParams)
+		},
+		SilenceUsage: true,
+	}
+
+	rootCmd.PersistentFlags().StringVar(&globalParams.ConfFilePath, flags.CfgPath, flags.DefaultConfPath, "Path to datadog.yaml config")
+
+	if flags.DefaultSysProbeConfPath != "" {
+		rootCmd.PersistentFlags().StringVar(&globalParams.SysProbeConfFilePath, flags.SysProbeConfig, flags.DefaultSysProbeConfPath, "Path to system-probe.yaml config")
+	}
+
+	rootCmd.PersistentFlags().StringVarP(&globalParams.PidFilePath, "pid", "p", "", "Path to set pidfile for process")
+	rootCmd.PersistentFlags().BoolVarP(&globalParams.Info, "info", "i", false, "Show info about running process agent and exit")
```

Comment:
We don't need to fix this now but should we deprecate the `info` parameter and move it to a subcommand?

Reply:
Agreed, `status` handles it.